lib/wahwah/mp4/atom.rb

```diff
@@ -53,6 +52,8 @@ def initialize
         # If the size field of an atom is set to 0, which is allowed only for a top-level atom,
         # designates the last atom in the file and indicates that the atom extends to the end of the file.
         @size = @file_io.size if @size == 0
+        return unless valid?
```

Comment:
Eventually, we do need to return back if it isn't valid, but I put that check near the end, after we attempted to assign the size if `size==0`